Public/Src/Cache/ContentStore/Distributed/NuCache/RocksDbContentLocationDatabase.cs

```diff
@@ -507,6 +529,162 @@ private byte[] GetKey(ShortHash hash)
             return hash.ToByteArray();
         }
 
+        #region Memoization
+        // TODO(jubayard): garbage collection / removal in general
+
+        /// <inheritdoc />
+        public override GetContentHashListResult GetContentHashList(OperationContext context, StrongFingerprint strongFingerprint)

```

Comment:
Please, use 'context.PerformOperation' for all the new methods and add new counter to track the duration.

Reply:
This is not an option for GetSelectors, since it may return a few results and the fail. What do you want me to do in this case?

---
In reply to: [302175307](https://github.com/microsoft/BuildXL/pull/571#discussion_r302175307) [](ancestors = 302175307)